azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageRampupServlet.java

```diff
@@ -0,0 +1,226 @@
+      final HttpServletResponse resp, final Session session) throws ServletException, IOException {
+    try {
+      final String jsonPayload = HttpRequestUtils.getBody(req);
+      // Check for required permission to invoke the API
+      final String imageType = JSONUtils
+          .extractTextFieldValueFromJsonString(jsonPayload, IMAGE_TYPE);
+      if (!hasPermission(imageType, session.getUser(), Type.CREATE)) {
+        throw new ImageMgmtInvalidPermissionException("Invalid permission to create image rampup "
+            + "plan");
+      }
+      // Build ImageMetadataRequest DTO to transfer the input request
+      final ImageMetadataRequest imageMetadataRequest = ImageMetadataRequest.newBuilder()
+          .jsonPayload(jsonPayload)
+          .user(session.getUser().getUserId())
+          .build();
+      // Create image version metadata and image version id
+      final Integer imageRampupPlanId = this.imageRampupService
+          .createImageRampupPlan(imageMetadataRequest);
+      // prepare to send response
+      resp.setStatus(HttpStatus.SC_CREATED);
+      resp.setHeader("Location",
+          CREATE_IMAGE_RAMPUP_URI_TEMPLATE.createURI(imageRampupPlanId.toString()));
+      sendResponse(resp, HttpServletResponse.SC_CREATED, new HashMap<>());
+    } catch (final ImageMgmtValidationException e) {
+      log.error("Input for creating image rampup plan is invalid", e);
+      sendErrorResponse(resp, HttpServletResponse.SC_BAD_REQUEST,
+          "Bad request for creating image rampup plan. " + e.getMessage());
+    } catch (final ImageMgmtInvalidPermissionException e) {
+      log.error("Unable to create image rampup plan. Invalid permission.", e);
+      sendErrorResponse(resp, HttpServletResponse.SC_FORBIDDEN,
+          "Unable to create image rampup plan. Reason: " + e.getMessage());
+    } catch (final Exception e) {
+      log.error("Exception while creating image rampup plan", e);
+      sendErrorResponse(resp, HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+          "Exception while creating image rampup plan. " + e.getMessage());
+    }
+  }
+
+  private void handleUpdateImageRampup(final HttpServletRequest req,
+      final HttpServletResponse resp, final Session session,
+      final Map<String, String> templateVariableToValue) throws ServletException,
+      IOException {
+    try {
+      final String imageType = templateVariableToValue.get(IMAGE_TYPE);
+      if (imageType == null) {
+        log.error("Image type can't be null. Must provide valid image type to update rampup.");
+        throw new ImageMgmtValidationException(
+            "Image type can't be null. Must provide valid image type to update rampup.");
+      }
+      // Check for required permission to invoke the API
+      if (!hasPermission(imageType, session.getUser(), Type.UPDATE)) {
+        throw new ImageMgmtInvalidPermissionException("Invalid permission to update image rampup "
```

Comment:
Can you please add debug log that user xyz does not have permission to update abc image type? Same for create and other actions.

Reply:
Done.

Reply:
Nit: Missing comma after user: %s

Reply:
Done.